### PR TITLE
Add tests

### PR DESCRIPTION
**Pull request by willemneal**

### CODE REVIEW

File: tests/asconfig/target/package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "private": true,
   "scripts": {
-    "test": "node ../index.js"
+    "test": "node ../index.js --target && node ../index.js --showConfig --runtime none"
```
Comment (dcodeIO) [inline tests/asconfig/target/package.json:4]: Is the `--target` without a target intended?

**Comment by dcodeIO** [comment]
Thanks, looks good! Just one question above.

File: tests/asconfig/target/package.json (hunk 1 above)
Comment (willemneal) [reply tests/asconfig/target/package.json]: Just fixed it!

File: cli/asc.js [hunk 2]
```diff
@@ -303,7 +303,7 @@ exports.main = function main(argv, options, callback) {
     // Append entries
     if (asconfig.entries) {
       for (let entry of asconfig.entries) {
-        argv.push(optionsUtil.resolvePath(entry, asconfigDir));
+        argv.push(path.relative(baseDir, optionsUtil.resolvePath(entry, asconfigDir)));
```
Comment (dcodeIO) [inline cli/asc.js:306]: Why is this necessary when `asconfigDir` is already relative to `baseDir`?

**Comment by dcodeIO** [comment]
Hmm, merging https://github.com/AssemblyScript/assemblyscript/pull/1411 into here looks like a mistake?

File: cli/asc.js (hunk 2 above)
Comment (willemneal) [reply cli/asc.js:306]: `resolvePath` returns an absolute path.  This was mainly to output a relative path when `--showConfig` is passed.

Comment (dcodeIO) [reply cli/asc.js:306]: It returns an absolute path only if `entry` is already absolute, where I guess keeping it that way makes sense for debugging purposes (so one knows about where the entry comes from by looking at the asconfigs), or node resolution is used, which is debatable. In the latter case, the place to change it would be inside of `resolvePath` where `require.resolve` is called, though.

Comment (dcodeIO) [reply cli/asc.js:306]: Oh, wait, for entries node resolution isn't used anyway, so the `relative` here seems redundant.

**Comment by willemneal** [comment]
I had forgotten to add the `node_modules` in the test for node resolution.

File: cli/asc.js (hunk 2 above)
Comment (willemneal) [reply cli/asc.js:306]: It can be used to get the asconfigDir, which returns an absolute path.

Comment (dcodeIO) [reply cli/asc.js:306]: What do you mean?

Comment (willemneal) [reply cli/asc.js:306]: When the extends field the file could be found with node resolution, which returns an absolute path, which means `path.join(entry, asconfigDir)` would return an absolute path.  But I just really want to get this finished so if you don't like it then I can take it out.

Comment (dcodeIO) [reply cli/asc.js:306]: Thanks for clarifying. I'm not necessarily opposed to changing node resolution to relative, but if we do so I think we should do it in `resolvePath` where `require.resolve` is used (and the path returned as absolute), so we don't break explicitly specified absolute paths given as the value of `extends`. Like, if a user specifies something absolute explicitly, I'd prefer to not mangle it but keep it as-is.

Comment (dcodeIO) [reply cli/asc.js:306]: Changed it in #1406 accordingly.

Comment (willemneal) [reply cli/asc.js:306]: very true.

Comment (willemneal) [reply cli/asc.js:306]: Just looked at your changes to #1406, and I think you should roll it back and just keep everything as is.  The issue was that I wanted the path relative to `baseDir` in asc.js, but in resolvePath, `baseDir` is passed in as `asconfigDir.` So we could adapt that to always pass in top level `baseDir` or just forget about it all together.

Comment (dcodeIO) [reply cli/asc.js:306]: Ok, let's remove the `path.relative` here then.

**Comment by dcodeIO** [comment]
As noted above, #1411, which is unrelated, somehow made it into this PR. With the original base branch now merged, I suggest to rebase this PR cleanly on master.

**Comment by willemneal** [comment]
I think I'll just close this and reopen it to master.
